python-avd/pyavd/_eos_designs/shared_utils/wan.py

```diff
@@ -475,28 +475,82 @@ def wan_ha_peer(self: SharedUtils) -> str | None:
             return self.switch_data_node_group_nodes[0]["name"]
         raise AristaAvdError("Unable to find WAN HA peer within same node group")
 
+    @cached_property
+    def configured_wan_ha_interfaces(self: SharedUtils) -> set:
+        """
+        Read the device wan_ha.ha_interfaces node settings
+        """
+        return get(self.switch_data_combined, "wan_ha.ha_interfaces", default=[])
+
+    @cached_property
+    def vrf_default_uplinks(self: SharedUtils) -> list:
+        """
+        Return the uplinkss in VRF default
+        """
+        return [uplink for uplink in self.get_switch_fact("uplinks") if get(uplink, "vrf") is None]
+
+    @cached_property
+    def vrf_default_uplink_interfaces(self: SharedUtils) -> list:
+        """
+        Return the uplink interfaces in VRF default
+        """
+        return [uplink["interface"] for uplink in self.vrf_default_uplinks]
+
+    @cached_property
+    def use_uplinks_for_wan_ha(self: SharedUtils) -> bool:
+        """
+        Return true or false
+        """
+        interfaces = set(self.configured_wan_ha_interfaces)
+        uplink_interfaces = set(self.vrf_default_uplink_interfaces)
+
+        if interfaces.issubset(uplink_interfaces):
+            return True
+        if not interfaces.intersection(uplink_interfaces):
+            if len(interfaces) > 1:
+                raise AristaAvdError("AVD does not support multiple HA interfaces when not using uplinks.")
+            return False
+        raise AristaAvdError("Either all `wan_ha.interfaces` must be uplinks interfaces or all of them must not be uplinks.")
```

Comment:
This error says wan_ha.interfaces, but the data model is wan_ha.ha_interfaces.
```suggestion
        raise AristaAvdError("Either all `wan_ha.ha_interfaces` must be uplink interfaces or all of them must not be uplinks.")
```